test/integration/model/bulk-create/include.test.js

```diff
@@ -621,6 +621,119 @@ describe(Support.getTestDialectTeaser('Model'), () => {
         expect(persistedUsers[1].jobs).to.be.ok;
         expect(persistedUsers[1].jobs.length).to.equal(2);
       });
+
+      it('should pass options to associated models as well', async function () {
+        const Player = this.sequelize.define('Player', {
+          code: {
+            type: DataTypes.STRING,
+            primaryKey: true,
+          },
+          name: DataTypes.STRING,
+          surname: DataTypes.STRING,
+          dateBirth: DataTypes.DATE,
+          country: DataTypes.STRING,
+        });
+
+        const Team = this.sequelize.define('Team', {
+          code: {
+            type: DataTypes.STRING,
+            primaryKey: true,
+          },
+          name: DataTypes.STRING,
+          country: DataTypes.STRING,
+          color: DataTypes.STRING,
+          trophies: DataTypes.INTEGER,
+        });
+
+        Team.hasMany(Player, { as: 'players' });
+        Player.belongsTo(Team);
+
+        await this.sequelize.sync();
+
+        const dialect = this.sequelize.options.dialect;
+
+        if (dialect === 'mssql' || dialect === 'db2') {
+          expect(
+            async () => await Team.bulkCreate({}).to.throw(
+              `${dialect} does not support the updateOnDuplicate option.`,
+            ),
+          );
+        } else {
+          await Team.bulkCreate(
+            [
+              {
+                code: 'team1',
+                name: 'nameTeam1',
+                country: 'countryTeam001',
+                color: 'colorTeam1',
+                trophies: 1,
+                players: [
+                  {
+                    code: 'player1',
+                    name: 'playername1',
+                    surname: 'playernSurname1',
+                    dateBirth: new Date(),
+                    country: 'country1',
+                  },
+                ],
+              },
+            ],
+            {
+              include: [
+                {
+                  model: Player,
+                  as: 'players',
+                  updateOnDuplicate: ['name'],
+                },
+              ],
+              updateOnDuplicate: ['country', 'color', 'trophies'],
+            },
+          );
+
+          let playerOfInterest = await Player.findOne({
+            where: { code: 'player1' },
+          });
+
+          expect(playerOfInterest.name).to.equal('playername1');
```

Comment:
we try to avoid re-using the same variable unless necessary. In this instance you could do this to limit the scope of this variable and allow using another variable with the same name further down:

```suggestion
          {
            const playerOfInterest = await Player.findOne({
              where: { code: 'player1' },
            });

            expect(playerOfInterest.name).to.equal('playername1');
          }
```